src/overlays/actors/ovl_En_Hy/z_en_hy.c

```diff
@@ -61,242 +69,380 @@ static ColliderCylinderInit D_80A71EC0 = {
-    { 0x00, 0.0f, 0.0f },   { 0x06, 20.0f, 10.0f }, { 0x06, 20.0f, 10.0f }, { 0x0A, 20.0f, 0.0f },
-    { 0x06, 20.0f, 10.0f }, { 0x06, 20.0f, 10.0f }, { 0x06, 20.0f, 10.0f }, { 0x06, 20.0f, 10.0f },
-    { 0x0A, 20.0f, 0.0f },
+    /* 0x00 */ u8 unkPresetIndex;
+    /* 0x04 */ f32 unkValueChild;
+    /* 0x08 */ f32 unkValueAdult;
+} EnHyInit1Info; // size = 0xC
+
+// indexed by EnHyType
+static EnHyInit1Info sInit1Infos[] = {
+    /*  0 */ { 0x06, 20.0f, 10.0f }, /*  1 */ { 0x06, 20.0f, 10.0f },
+    /*  2 */ { 0x07, 40.0f, 20.0f }, /*  3 */ { 0x06, 20.0f, 10.0f },
+    /*  4 */ { 0x07, 40.0f, 20.0f }, /*  5 */ { 0x08, 0.0f, -20.0f },
+    /*  6 */ { 0x09, 20.0f, 0.0f },  /*  7 */ { 0x09, 20.0f, 0.0f },
+    /*  8 */ { 0x06, 20.0f, 10.0f }, /*  9 */ { 0x06, 20.0f, 10.0f },
+    /* 10 */ { 0x06, 20.0f, 10.0f }, /* 11 */ { 0x06, 20.0f, 10.0f },
+    /* 12 */ { 0x00, 0.0f, 0.0f },   /* 13 */ { 0x06, 20.0f, 10.0f },
+    /* 14 */ { 0x06, 20.0f, 10.0f }, /* 15 */ { 0x0A, 20.0f, 0.0f },
+    /* 16 */ { 0x06, 20.0f, 10.0f }, /* 17 */ { 0x06, 20.0f, 10.0f },
+    /* 18 */ { 0x06, 20.0f, 10.0f }, /* 19 */ { 0x06, 20.0f, 10.0f },
+    /* 20 */ { 0x0A, 20.0f, 0.0f },
 };
 
 typedef struct {
-    /* 0x00 */ f32 unk_0;
-    /* 0x04 */ Vec3f unk_4;
-    /* 0x10 */ f32 unk_10;
-    /* 0x14 */ s8 unk_14;
-    /* 0x18 */ f32 unk_18;
-} EnHyUnknownStruct6; // size = 0x1C
-
-static EnHyUnknownStruct6 D_80A725A4[] = {
-    { 36.0f, { 0.0f, 0.0f, 600.0f }, 0.01f, 0x06, 30.0f },     { 40.0f, { -100.0f, 0.0f, 400.0f }, 0.01f, 0x06, 30.0f },
-    { 22.0f, { 0.0f, 0.0f, -200.0f }, 0.01f, 0x06, 30.0f },    { 20.0f, { -100.0f, 0.0f, 0.0f }, 0.01f, 0x06, 30.0f },
-    { 22.0f, { 0.0f, 0.0f, 0.0f }, 0.01f, 0x06, 30.0f },       { 21.0f, { 0.0f, 0.0f, 0.0f }, 0.01f, 0x06, 30.0f },
-    { 25.0f, { -100.0f, 0.0f, 600.0f }, 0.01f, 0x06, 30.0f },  { 28.0f, { -100.0f, 0.0f, 800.0f }, 0.01f, 0x06, 30.0f },
-    { 17.0f, { 0.0f, 0.0f, 700.0f }, 0.01f, 0x06, 30.0f },     { 18.0f, { 0.0f, 0.0f, 100.0f }, 0.01f, 0x06, 30.0f },
-    { 18.0f, { 0.0f, 0.0f, -200.0f }, 0.01f, 0x06, 30.0f },    { 17.0f, { 0.0f, 0.0f, 700.0f }, 0.01f, 0x06, 30.0f },
-    { 21.0f, { 0.0f, 0.0f, -300.0f }, 0.01f, 0x06, 30.0f },    { 20.0f, { 0.0f, 0.0f, -200.0f }, 0.01f, 0x06, 30.0f },
-    { 18.0f, { -200.0f, 0.0f, -200.0f }, 0.01f, 0x06, 30.0f }, { 27.0f, { -100.0f, 0.0f, 800.0f }, 0.01f, 0x06, 30.0f },
-    { 19.0f, { 400.0f, 0.0f, 0.0f }, 0.01f, 0x04, 30.0f },     { 19.0f, { 400.0f, 0.0f, 0.0f }, 0.01f, 0x06, 30.0f },
-    { 40.0f, { -100.0f, 0.0f, 400.0f }, 0.01f, 0x06, 30.0f },  { 17.0f, { 0.0f, 0.0f, 700.0f }, 0.01f, 0x06, 30.0f },
-    { 20.0f, { 0.0f, 0.0f, -200.0f }, 0.01f, 0x06, 30.0f },
+    /* 0x00 */ f32 shadowScale;
+    /* 0x04 */ Vec3f modelOffset;
+    /* 0x10 */ f32 scale;
+    /* 0x14 */ s8 targetMode;
+    /* 0x18 */ f32 unkRange;
+} EnHyInit2Info; // size = 0x1C
+
+// indexed by EnHyType
+static EnHyInit2Info sInit2Infos[] = {
```

Comment:
might be an unpopular opinion but i dont think we need to note what its indexed by when context makes that clear